[FEAT] users, questions, results API 추가

pje1740 opened the pull request:
users, questions, results API 추가

pje1740 committed:
[CHORE] add eslint, prettier config files

pje1740 committed:
[CHORE] install eslint, mysql2 modules

pje1740 committed:
[FEAT] add questions, results, users router

pje1740 committed:
[FEAT] add router for getting questions and choices

pje1740 committed:
[FEAT] add router for adding user result to the DB

pje1740 committed:
[FEAT] add router for retrieving total number of users

pje1740 committed:
[FEAT] connect mysql2 db and create db pool

pje1740 committed:
[FIX] delete view, public files

jwbeen reviewed (changes requested):
questions를 호출하면 마지막 문제의 데이터만 받아옵니다.
전체 문제의 데이터를 받아오게 수정 부탁드립니다.
그리고 main branch가 아니라 develop branch로 merge해주세요! (바꿔놨어요~)

pje1740 committed:
[FIX] fix the logic inside retreiving the questions data

Changed reduce function to map and use Promise.all to resolve all the promises.
Reduce returns a single value while map returns a new array.

jwbeen commented:
LGTM!
